### ISSUE TITLE
Incompatibility with kotlinx.coroutines 0.16

### ISSUE THREAD

**Issue by LouisCAD** [gildor/kotlin-coroutines-retrofit#16]
Hi,

When using this library with kotlinx.coroutines version 0.16, I get the following error:

```stacktrace
java.lang.NoSuchMethodError: No direct method <init>(Lkotlin/coroutines/experimental/Continuation;Z)V in class Lkotlinx/coroutines/experimental/CancellableContinuationImpl; or its super classes (declaration of 'kotlinx.coroutines.experimental.CancellableContinuationImpl' appears in /data/app/my.app.packagename.debug-2/base.apk)
                                                                                             at ru.gildor.coroutines.retrofit.CallAwaitKt.awaitResult(CallAwait.kt:124)
```

The issue may be present with version 0.15 too

**Comment by gildor** [comment]
I will release version with kotlinx.coroutines 0.16 tomorrow

**Comment by LouisCAD** [comment]
Is an issue holding you back from releasing the update?

**Comment by LouisCAD** [comment]
Fixed by #15, #17, #18 and #19!
